pkg/controller/servicebindingrequest/nested/nested.go

```diff
@@ -0,0 +1,188 @@
+package nested
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/redhat-developer/service-binding-operator/pkg/controller/servicebindingrequest/nested/acc"
+)
+
+// getValueFromMap attempts to retrieve from `obj` a value from the given path
+// `path`.
+func getValueFromMap(obj map[string]interface{}, path Path) (interface{}, bool, error) {
+	head, exists := path.Head()
+	if !exists {
+		return obj, true, nil
+	}
+	val, exists := obj[head.Name]
+	if !exists {
+		return nil, false, nil
+	}
+
+	return getValue(val, path.Tail())
+}
+
+// GetValueFromMap attempts to retrieve from `obj` a value from the given path
+// `p`.
+func GetValueFromMap(obj map[string]interface{}, p string) (interface{}, bool, error) {
```

Comment:
Double check whether this is still needed and move this code to a library in another repository if not.